hack/internal/tools/kueuectl-docs/generators/doc.go

```diff
@@ -0,0 +1,374 @@
+		parts := strings.Split(cmd.CommandPath(), " ")
+		if len(parts) == 2 {
+			indexFile = true
+			fullname = "_index.md"
+		}
+		fullname = filepath.Join(subdir, fullname)
+	}
+	filename := filepath.Join(outputDir, fullname)
+
+	f, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	if _, err := io.WriteString(f, filePrepender(filename)); err != nil {
+		return err
+	}
+	if err := GenReference(cmd, templatesDir, f, linkHandler, indexFile); err != nil {
+		return err
+	}
+	return nil
+}
+
+func GenReference(cmd *cobra.Command, templatesDir string, w io.Writer, linkHandler func(string) string, indexFile bool) error {
+	ref := getReference(cmd, linkHandler, indexFile)
+
+	tmpl, err := template.New("main.html").
+		Funcs(template.FuncMap{
+			"html":  htmlFuncMap,
+			"usage": usageFuncMap,
+		}).
+		ParseGlob(fmt.Sprintf("%s/*", templatesDir))
+	if err != nil {
+		return err
+	}
+	if err := tmpl.Execute(w, ref); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func htmlFuncMap(s string) template.HTML {
+	return template.HTML(s)
+}
+
+func usageFuncMap(usage string) template.HTML {
+	usage = html.EscapeString(usage)
+	usage = strings.TrimSuffix(usage, "\\n")
+	usage = strings.ReplaceAll(usage, "\\n", "<br/>")
+	return htmlFuncMap(usage)
+}
+
+func getReference(cmd *cobra.Command, linkHandler func(string) string, indexFile bool) *Reference {
+	cmd.InitDefaultHelpCmd()
+	cmd.InitDefaultHelpFlag()
+
+	ref := &Reference{
+		NoList:           indexFile,
+		Title:            strings.TrimSpace(cmd.CommandPath()),
+		Description:      cmd.Long,
+		Example:          cmd.Example,
+		Options:          getOptions(cmd.NonInheritedFlags()),
+		InheritedOptions: getOptions(cmd.InheritedFlags()),
+		Links:            getLinks(cmd, linkHandler, indexFile),
+	}
+
+	if len(ref.Description) == 0 {
+		ref.Description = cmd.Short
+	}
+
+	if cmd.Runnable() {
+		ref.UseLine = cmd.UseLine()
+	}
+
+	return ref
+}
+
+func getOptions(f *pflag.FlagSet) []Option {
+	var options []Option
+
+	f.VisitAll(func(flag *pflag.Flag) {
+		if len(flag.Deprecated) > 0 || flag.Hidden {
+			return
+		}
+
+		option := Option{
+			Name:        flag.Name,
+			Usage:       flag.Usage,
+			ValueType:   getValueType(flag),
+			DefVal:      getDefaultVal(flag),
+			NoOptDefVal: getNoOptDefVal(flag),
+		}
+
+		if len(flag.ShorthandDeprecated) == 0 {
+			option.Shorthand = flag.Shorthand
+		}
+
+		options = append(options, option)
+	})
+
+	return options
+}
+
+func getValueType(flag *pflag.Flag) string {
+	name := flag.Value.Type()
+	switch name {
+	case "bool":
+		return ""
+	case "float64", "float32":
+		return "float"
+	case "int64", "severity", "Level":
+		return "int"
+	case "uint64":
+		return "uint"
+	case "stringSlice", "stringArray":
+		return "strings"
+	case "intSlice":
+		return "ints"
+	case "uintSlice":
+		return "uints"
+	case "boolSlice":
+		return "bools"
+	case "mapStringString":
+		return "&lt;comma-separated 'key=value' pairs&gt;"
```

Comment:
Do we still need to escape this if we are using templates?

Reply:
Thanks